opentracing/src/main/java/opentracing/Tracer.java

```diff
@@ -13,25 +13,41 @@
  */
 package opentracing;
 
+
 /**
- * Tracer is a simple, thin interface for Span creation.
+ * Tracer is a simple, thin interface for Span creation, and Span propagation into different transport formats.
  */
 public interface Tracer {
 
   /**
-   * Create, start, and return a new Span with the given `operationName`, all without specifying a
-   * parent Span that can be used to incorporate the newly-returned Span into an existing trace.
+   * Create, start, and return a new Span with the given `operationName`.
+   * An optional parent Span can be specified used to incorporate the newly-returned Span into an existing trace.
    *
    * <p>Example:
    * <pre>{@code
    * Tracer tracer = ...
-   * Span feed = tracer.startTrace("GetFeed");
-   * Span http = tracer.startTrace("HandleHTTPRequest")
+   * Span feed = tracer.startTrace("GetFeed", null);
+   * Span http = tracer.startTrace("HandleHTTPRequest", feed)
    *                   .setTag("user_agent", req.UserAgent)
    *                   .setTag("lucky_number", 42);
    * }</pre>
    */
-  Span startTrace(String operationName);
+  Span startSpan(String operationName, /* @Nullable */ Span parent);
+
+  /** Takes two arguments:
+   *    a Span instance, and
+   *    a “carrier” object in which to inject that Span for cross-process propagation.
+   *
+   */
+  <T> void inject(Span span, T carrier);
```

Comment:
I see that the controversial carriers are no longer part of this PR -- works for me. :)  I like incremental PRs and it's great if we can merge the less controversial parts first.

That said, we do have a bit of a fork-in-the-road re the "carriers"... regardless of the Split/NonSplit stuff we have been discussing, is it realistic to say that a `byte[]` is *always* a specific carrier type? I would argue that it's certainly not realistic... so we will end up with a situation where thin classes need to wrap around things like a `byte[]` in order to get the necessary type information into `inject`. Not the end of the world, but I do want to make sure this is a deliberate choice... in other languages we have been using explicit identifiers/strings/whatever to determine the type/semantics of the carrier parameter.